schedule-update/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	dispatch::Weight,
+	traits::Get,
+	weights::{DispatchClass, GetDispatchInfo},
+	Parameter,
+};
+use frame_system::{self as system, ensure_root, ensure_signed};
+use sp_runtime::{
+	traits::{Dispatchable, One},
+	DispatchError, RuntimeDebug,
+};
+use sp_std::{prelude::*, result};
+
+mod mock;
+mod tests;
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug)]
+pub enum DelayedDispatchTime<BlockNumber> {
+	At(BlockNumber),
+	After(BlockNumber),
+}
+
+type DispatchId = u32;
+type CallOf<T> = <T as Trait>::Call;
+
+pub trait Trait: frame_system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+	type Call: Parameter + Default + Dispatchable<Origin = <Self as frame_system::Trait>::Origin> + GetDispatchInfo;
+	type MaxScheduleDispatchWeight: Get<Weight>;
+}
+
+decl_event!(
+	/// Event for schedule-update module.
+	pub enum Event<T> where
+		<T as frame_system::Trait>::BlockNumber,
+	{
+		/// Add schedule dispatch success (BlockNumber, DispatchId)
+		ScheduleDispatch(BlockNumber, DispatchId),
+		/// Cancel deplayed dispatch success (DispatchId)
+		CancelDeplayedDispatch(DispatchId),
+		/// Schedule dispatch success (BlockNumber, DispatchId)
+		ScheduleDispatchSuccess(BlockNumber, DispatchId),
+		/// Schedule dispatch failed (DispatchId, DispatchError)
+		ScheduleDispatchFail(DispatchId, DispatchError),
+	}
+);
+
+decl_error! {
+	/// Error for schedule-update module.
+	pub enum Error for Module<T: Trait> {
+		BadOrigin,
+		CannotGetNextId,
+		NoPermission,
+		DispatchNotExisted,
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as ScheduleUpdate {
+		pub NextId get(fn next_id): DispatchId;
+		pub DelayedNormalDispatches get(fn delayed_normal_dispatches): double_map hasher(blake2_256) T::BlockNumber, hasher(blake2_256) DispatchId => (Option<T::AccountId>, CallOf<T>, DispatchId);
```

Comment:
```suggestion
		pub DelayedNormalDispatches get(fn delayed_normal_dispatches): double_map hasher(twox_64_concat) T::BlockNumber, hasher(twox_64_concat) DispatchId => (Option<T::AccountId>, CallOf<T>, DispatchId);
```

Avoid using `hasher(blake2_256)`, use `hasher(twox_64_concat)` for keys that user cannot arbitrarily choose, and use `blake2_128_concat` otherwise. 

Later we can iterate with keys and values then we can remove `DispatchId` from value.